src/app/main/component/events/services/events.service.ts

```diff
@@ -11,9 +11,36 @@ export class EventsService implements OnDestroy {
   public currentForm: PagePreviewDTO;
   private backEnd = environment.backendLink;
   private destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
+  private isAddressFillSubject: BehaviorSubject<boolean[]> = new BehaviorSubject<boolean[]>([]);
 
   constructor(private http: HttpClient) {}
 
+  public setIsAddressFill(dates: DateEvent[], submit?: boolean, init?: boolean, check?: boolean, ind?: number): void {
+    const currentValues = this.isAddressFillSubject.getValue();
+    let newArray;
+
+    if (init) {
+      newArray = [];
+    } else if (submit) {
+      newArray = dates.map((nextValue) => !(nextValue.coordinatesDto.latitude || nextValue.onlineLink));
+    } else if (check && ind !== undefined) {
```

Comment:
check && ind

Reply:
I used this statement because the ind could be 0, 1, 2, 3... I check whether I have some index or not. Is it better to use "check && ind !== null" here?